contracts/ERC20Indexable.sol

```diff
@@ -18,45 +19,51 @@ contract ERC20Indexable is IERC20Indexable, ERC20, PositionManagerDependent {
 
     // --- Variables ---
 
-    uint256 public override currentIndex;
+    uint256 internal storedIndex;
 
     // --- Constructor ---
 
     constructor(
         address positionManager_,
         string memory name_,
-        string memory symbol_
+        string memory symbol_,
+        uint256 cap_
     )
         ERC20(name_, symbol_)
+        ERC20Capped(cap_)
         PositionManagerDependent(positionManager_)
     {
-        currentIndex = INDEX_PRECISION;
+        storedIndex = INDEX_PRECISION;
         emit ERC20IndexableDeployed(positionManager_);
     }
 
     // --- Functions ---
 
-    function mint(address to, uint256 amount) external override onlyPositionManager {
-        _mint(to, amount.divUp(currentIndex));
+    function mint(address to, uint256 amount) public virtual override onlyPositionManager {
+        _mint(to, amount.divUp(storedIndex));
     }
 
-    function burn(address from, uint256 amount) external override onlyPositionManager {
-        _burn(from, amount == type(uint256).max ? ERC20.balanceOf(from) : amount.divUp(currentIndex));
+    function burn(address from, uint256 amount) public virtual override onlyPositionManager {
+        _burn(from, amount == type(uint256).max ? ERC20.balanceOf(from) : amount.divUp(storedIndex));
     }
 
-    function setIndex(uint256 backingAmount) external override onlyPositionManager {
+    function setIndex(uint256 backingAmount) public virtual override onlyPositionManager {
         uint256 supply = ERC20.totalSupply();
```

Comment:
Maybe would look better that this part
```
uint256 supply = ERC20.totalSupply();
        uint256 newIndex = (backingAmount == 0 && supply == 0) ? INDEX_PRECISION : backingAmount.divUp(supply);
        storedIndex = newIndex;
        emit IndexUpdated(newIndex);
 ```

put into internal `_setIndex` function, and after that, from `InterestRateDebtToken` call `_setIndex` instead of `super.setIndex()`. Also, because of that, in `InterestRateDebtToken` and function `setIndex` put modifier `onlyPositionManager`.